connectors/sources/onedrive.py

```diff
@@ -529,6 +584,93 @@ def prepare_doc(self, file):
             "url": file.get("webUrl"),
         }
 
+    def _dls_enabled(self):
+        if self._features is None:
+            return False
+
+        if not self._features.document_level_security_enabled():
+            return False
+
+        return self.configuration["use_document_level_security"]
+
+    async def _decorate_with_access_control(self, document, user_id, file_id):
+        if self._dls_enabled():
+            entity_permissions = await self.get_entity_permission(
+                user_id=user_id, file_id=file_id
+            )
+            document[ACCESS_CONTROL] = list(
+                set(document.get(ACCESS_CONTROL, []) + entity_permissions)
+            )
+        return document
+
+    async def _user_access_control_doc(self, user):
+        email = user.get("mail")
+        username = user.get("userPrincipalName")
+
+        prefixed_email = _prefix_email(email)
+        prefixed_username = _prefix_user(username)
+        prefixed_user_id = _prefix_user_id(user.get("id"))
+
+        prefixed_groups = set()
+        user_groups = user.get("transitiveMemberOf", [])
+        if len(user_groups) < 100:  # $expand param has a max of 100
+            for group in user_groups:
+                prefixed_groups.add(_prefix_group(group.get("id")))
+        else:
+            async for group in self.client.list_groups(user_id=user.get("id")):
+                prefixed_groups.add(_prefix_group(group.get("id")))
+
+        access_control = list(
+            {prefixed_email, prefixed_username, prefixed_user_id}.union(prefixed_groups)
+        )
+        return {
+            "_id": email if email else username,
+            "identity": {
+                "email": prefixed_email,
+                "username": prefixed_username,
+                "user_id": prefixed_user_id,
+            },
+            "created_at": user.get("createdDateTime", iso_utc()),
+        } | es_access_control_query(access_control)
+
+    async def get_access_control(self):
+        if not self._dls_enabled():
+            self._logger.warning("DLS is not enabled. Skipping")
+            return
+
+        self._logger.info("Fetching all users")
+        async for user in self.client.list_users(include_groups=True):
+            yield await self._user_access_control_doc(user=user)
+
+    async def get_entity_permission(self, user_id, file_id):
+        if not self._dls_enabled():
+            return []
+
+        def _get_id(permission, identity):
+            if identity not in permission:
+                return
```

Comment:
With the change, you don't need these two lines.

Actually, I don't feel it's necessary to define this function, and it's a bit confusing, because you pass in `identity` to param `permission`:
```python
user_permission = _get_id(identity, "user")
group_permission = _get_id(identity, "group")
```

I think you can simply remove this function, and do
```python
user_permission = identity.get("user", {}).get("id")
group_permission = identity.get("user", {}).get("group")
```